### PR TITLE
Enable auto backup google#63

### PR DESCRIPTION
**Pull request by TheBonheurs**
- Relevant Issues: #63 
- Related Pull Requests: -

* * *

## What
Auto backup feature for apps provided by google

## Why
This Pull Request is needed because now we can assure the users their data will be backed up.

## How
This feature can be viewed/tested within the project by :

- The user has enabled backup on the device in Settings > Backup & Reset.
- At least 24 hours have elapsed since the last backup.
- The device is idle and charging.
- The device is connected to a Wi-Fi network. If the device is never connected to a wifi network, then Auto Backup never occurs.

## Alternative implementation
Other implementations that I've have considered are casting our contacts to google contacts and using that specific google contact backup feature but according to @jverbraeken it was too complicated.

### CODE REVIEW

**Comment by ericcornelissen** [comment]
I have one important question about this feature, is it enabled by default?

I don't think our users, who are particularly nervous about their privacy, want to have this feature enabled by default... I would be okay with a checkbox in the first time use activity asking users if they want to use this feature.

File: app/src/main/java/com/nervousfish/nervousfish/modules/pairing/INfcHandler.java [hunk 1]
```diff
@@ -5,5 +5,5 @@
 /**
  * Defines a module used for pairing using NFC that can be used by a service locator.
  */
-public interface INfcHandler extends IPairingHandler, IModule {
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/modules/pairing/INfcHandler.java]: Why are you changing this in this PR?

File: app/src/main/java/com/nervousfish/nervousfish/modules/pairing/IQRHandler.java [hunk 2]
```diff
@@ -6,5 +6,5 @@
  * Defines a module used for pairing using a QR code that can be used by a service locator.
  */
 
-public interface IQRHandler extends IPairingHandler, IModule {
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/modules/pairing/IQRHandler.java]: Why are you changing this in this PR?

File: app/src/main/java/com/nervousfish/nervousfish/modules/pairing/INfcHandler.java (hunk 1 above)
Comment (TheBonheurs) [inline app/src/main/java/com/nervousfish/nervousfish/modules/pairing/INfcHandler.java]: No idea, I've added them manually again...

File: app/src/main/java/com/nervousfish/nervousfish/modules/pairing/IQRHandler.java (hunk 2 above)
Comment (TheBonheurs) [inline app/src/main/java/com/nervousfish/nervousfish/modules/pairing/IQRHandler.java]: Idem

**Comment by TheBonheurs** [comment]
As to your question yes if you have Google backup enabled on your mobile device this will happen automatically. But I think if the users find this sketchy they will have that option disabled on their device. This will mean that their data won't be backed up though bu as I said in the alternatives I haven't found an elegant way to perform this backup manually.

**Comment by ericcornelissen** [comment]
OK, just to be clear: If I don't have Google Backup enabled, then this feature won't do anything?

If that is the case I'll approve this PR 😄

**Comment by TheBonheurs** [comment]
knipsel

**Comment by ericcornelissen** [comment]
@TheBonheurs, why did you close this PR?

**Comment by TheBonheurs** [comment]
No idea tbh @ericcornelissen , can't remember it either

**Comment by TheBonheurs** [comment]
It could be that I moved this item on the waffle queue to done or something like that but it shouldn't immediately close this right?

**Comment by ericcornelissen** [comment]
I think it might, it seems like a plausible explanation 😆

**Comment by ericcornelissen** [comment]
Reopening this, so we can reconsider merging it 😄

**Pull request merged**